tide/master/PixelStreamWindowManager.cpp

```diff
@@ -111,9 +111,20 @@ void PixelStreamWindowManager::openWindow( const QString& uri,
 
     const auto type = _isPanel( uri ) ? ContentWindow::PANEL :
                                         ContentWindow::DEFAULT;
-    ContentPtr content = (stream == PixelStreamWindowManager::WEBBROWSER) ?
-                                    ContentFactory::getWebbrowserContent( uri ) :
-                                    ContentFactory::getPixelStreamContent( uri );
+
+    ContentPtr content;
```

Comment:
Hmmm, creating the appropriate type of Content should be the responsibility of the ContentFactory. There was already an exception with the getWebbrowserContent(), but this is becoming too many exceptions. I think we should move the StreamType to ContentType, and pass this as the second parameter to ContentFactory::getPixelStreamContent and remove getWebbrowserContent.